gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/resources/auth/provider/ClientPostAuthProvider.java

```diff
@@ -54,10 +64,31 @@ public void handle(Client client, RoutingContext context, Handler<AsyncResult<Cl
         final String clientId = getClientId(context.request());
         final String clientSecret = getClientSecret(context.request());
 
-        if (!client.getClientId().equals(clientId) || !client.getClientSecret().equals(clientSecret)) {
+        if (!client.getClientId().equals(clientId)) {
             handler.handle(Future.failedFuture(new InvalidClientException(ClientAuthHandler.GENERIC_ERROR_MESSAGE)));
             return;
         }
+
+        if (!isEmpty(client.getClientSecrets())) {
+            // take the first one as for now, we do not manage multiple secrets
+            ClientSecret hashedSecret = client.getClientSecrets().get(0);
+            var pwdEncoder = client.getSecretSettings()
+                    .stream().filter(settings -> settings.getId().equals(hashedSecret.getSettingsId()))
+                    .findFirst()
+                    .map(appSecretService::getOrCreatePasswordEncoder)
+                    .orElseGet(() -> appSecretService.getOrCreateNoOpPasswordEncoder());
```

Comment:
to remove codesmell:
.orElseGet(appSecretService::getOrCreateNoOpPasswordEncoder);